bin/dearchive

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+# AES SHA256 CBC 復号スクリプト (UTF-8)
+# =======================================
+#
+#
+# OpenSSL の AES-256-CBC と「共通鍵」を使って暗号化されたファイルを復号します。
+#
+# - 使い方の例：
+#   以下のコマンドで復号＆解凍済みファイル `myfile` が作成されます。
+#   （password.passwdはパスワードが書かれたテキストファイルです）
+#       $ ./dearchive myfile.tar.gz.aes myfile password.passwd
+#
+#   ４つ目の引数として秘密鍵を指定すると、３つ目の引数で渡したパスワードファイルがencスクリプトで暗号化
+#   されているとみなし、秘密鍵で復号してからパスワードとして利用します。
+#   （password.passwd.encはencスクリプトで暗号化済みのパスワードが記録されたバイナリファイルです）
+#       $ ./dearchive myfile.tar.gz.aes myfile password.passwd.enc ~/.ssh/github_key
+#
+# - 注意：利用前にスクリプトに実行権限を与えるのを忘れないでください。
+#
+
+# ヘルプ表示
+# ----------
+if [[ $# -lt 3 ]]; then
+    echo
+    echo "ファイルを復号して解凍します。（AES-256-CBC暗号）"
+    echo
+    echo "使い方: $0 <input file> <output file> <password file> [<private key>]"
+    echo
+    echo "- <input file>  : 復号＆解凍したいファイルのパス"
+    echo "- <output file> : 復号＆解凍済みファイルの出力先のパス"
+    echo "- <password file> : パスワードが記載されたファイル。"
+    echo "                    オプションの<private key>が指定されている場合は"
+    echo "                    パスワードがencスクリプトによって暗号化されているとみなします。"
+    echo
+    echo "[オプション]"
+    echo "- <private key> : encスクリプトで暗号化されたパスワードを復号するための秘密鍵。"
+    echo
+    exit 1
+fi
+
+md5s() {
+    if [ -e "$(which md5sum)" ]; then
+        echo "$1" | md5sum | awk '{ print $1 }'
+    elif [ -e "$(which md5)" ]; then
+        md5 -q -s "$1"
+    fi
+}
+
+# コマンド引数取得
+# ----------------
+INPUTFILE=$1
+OUTPUTFILE=$2
+PASSWORDFILE=$3
+PRIVATEKEYFILE=$4
+
+# 一時ファイル
+# ------------
+TMP=$(md5s $RANDOM)
+OUTPUTFILENAME=`basename ${OUTPUTFILE}`
```

Comment:
shellcheck の lint にいくつか引っ掛かっているようです。

* SC2006 バッククオートではなく、 `$( ... )` 形式を使う
* SC2086, SC2046 変数展開の際、ダブルクォートでくくる
* SC2064 trap の場合にシングルクォートでくくる（そうでないと `*` の展開が、trap 定義時に行われる）

お手数ですが、ご確認ください。

vscode でコンテナを使えば、shellcheck が入った環境になると思います。

あるいはコンテナが使えない環境であれば、web 上で shellcheck ができるサイトもあります → [shellcheck.net](https://www.shellcheck.net/)

```
Line 60:
OUTPUTFILENAME=`basename ${OUTPUTFILE}`
               ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.
                         ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2006, apply all SC2086)
OUTPUTFILENAME=$(basename "${OUTPUTFILE}")
 
Line 68:
trap "rm -rf /tmp/QiiCipher_${TMP}.*" 0
                            ^-- SC2064: Use single quotes, otherwise this expands now rather than when signalled.
 
Line 74:
    dec $PRIVATEKEYFILE $PASSWORDFILE $DECRYPTED_PASSWORDFILE
        ^-- SC2086: Double quote to prevent globbing and word splitting.
                        ^-- SC2086: Double quote to prevent globbing and word splitting.
                                      ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
    dec "$PRIVATEKEYFILE" "$PASSWORDFILE" "$DECRYPTED_PASSWORDFILE"
 
Line 75:
    PASSWORD=`cat ${DECRYPTED_PASSWORDFILE}`
             ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.
                  ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2006, apply all SC2086)
    PASSWORD=$(cat "${DECRYPTED_PASSWORDFILE}")
 
Line 77:
    PASSWORD=`cat ${PASSWORDFILE}`
             ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.
                  ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2006, apply all SC2086)
    PASSWORD=$(cat "${PASSWORDFILE}")
 
Line 87:
    -k $PASSWORD \
       ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
    -k "$PASSWORD" \
 
Line 94:
tar -xf "$TEMPFILE" -C `dirname ${OUTPUTFILE}`
                       ^-- SC2046: Quote this to prevent word splitting.
                       ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.
                                ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2006, apply all SC2086)
tar -xf "$TEMPFILE" -C $(dirname "${OUTPUTFILE}")
```